include/armpp/hal/registers.hpp

```diff
@@ -668,4 +668,221 @@ concept register_field = is_register_field<T>::value;
 template <typename T>
 concept readable_field = register_field<T> and requires(T const& reg) { reg.get(); };
 
+namespace detail {
+
+template <concepts::register_value T, std::size_t Size>
+struct array_field_accessor_base {
+    using value_type           = T;
+    static constexpr auto mask = util::bit_mask_v<0, Size>;
+
+public:
+    array_field_accessor_base() noexcept = default;
+    array_field_accessor_base(raw_register volatile& reg, std::size_t offset) noexcept
+        : reg_{&reg}, offset_{offset}
+    {}
+
+    array_field_accessor_base(array_field_accessor_base const&)     = delete;
+    array_field_accessor_base(array_field_accessor_base&&) noexcept = default;
+
+    array_field_accessor_base&
+    operator=(array_field_accessor_base const&)
+        = delete;
+    array_field_accessor_base&
+    operator=(array_field_accessor_base&&) noexcept
+        = default;
```

Comment:
**Suggestion:** Consider adding a default value for the `reg_` member of `array_field_accessor_base` to ensure it's never uninitialized. This will prevent potential undefined behavior.
```suggestion
    array_field_accessor_base() noexcept : reg_{nullptr}, offset_{0} {}
    array_field_accessor_base(raw_register volatile& reg, std::size_t offset) noexcept
        : reg_{&reg}, offset_{offset}
    {}

    array_field_accessor_base(array_field_accessor_base const&)     = delete;
    array_field_accessor_base(array_field_accessor_base&&) noexcept = default;

    array_field_accessor_base&
    operator=(array_field_accessor_base const&)
        = delete;
    array_field_accessor_base&
    operator=(array_field_accessor_base&&) noexcept
        = default;
```